### PR TITLE
cbuffer/cgetbuffer/caddbuffer doesn't accept a range

### LINKED ISSUE #14638: cbuffer/cad doesn't take visual selection as range '<,'>cad
### Steps to reproduce

select something
type  '<,'>cad
=> E16: Invalid range

### Expected behaviour

Should take visual range (like it takes regular range such as 1cad).
Btw in the issue #2955 you can see it's advised by someone to do such things as '<,'>cgetbuffer (very useful because as explained in the issue when you have a terminal running inside vim you regularly want to take the output of the command and put it in the fixlist)

### Version of Vim

VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Mar 14 2024 09:05:11) Included patches: 1-579, 1969, 580-1848, 4975, 5016, 5023, 5072, 2068, 1849-1854, 1857, 1855-1857, 1331, 1858, 1858-1859, 1873, 1860-1969, 1992, 1970-1992, 2010, 1993-2068, 2106, 2069-2106, 2108, 2107-2109, 2109-3995, 4563, 4646, 4774, 4895, 4899, 4901, 4919, 213, 1840, 1846-1847, 2110-2112, 2121 

### Environment

xterm-256color
WSL 2 ubuntu 22.04
windows terminal

### Logs and stack traces

_No response_

### PR DESCRIPTION
**Pull request by chrisbra**
Problem:  cbuffer/cgetbuffer/caddbuffer doesn't accept a range
          (ilan-schemoul after 8.1.1241)
Solution: Define ex commands with ADDR_LINES instead of ADDR_OTHER

fixes: #14638

### CODE REVIEW

File: src/testdir/test_quickfix.vim [hunk 1]
```diff
@@ -6496,4 +6496,70 @@ func Test_efm_format_b()
...
+  call assert_true(len(l) == 1 &&
+	\ l[0].lnum == 700 && l[0].col == 10 && l[0].text ==# 'Line 700')
+
+  enew!
+  let lines =<< trim END
+    Xtestfile9:900:55:Line 900
+    Xtestfile10:950:66:Line 950
+  END
+  silent! call setline(1, lines)
+  if (a:cchar == 'c')
```
Comment (yegappan) [inline src/testdir/test_quickfix.vim:6528]: Instead of this if-else, can you use `Xgetbuffer` here?

File: src/testdir/test_quickfix.vim [hunk 2]
```diff
@@ -6496,4 +6496,70 @@ func Test_efm_format_b()
...
+      call assert_fails('900,999caddbuffer', 'E16:')
+  else
+      call assert_fails('900,999laddbuffer', 'E16:')
+  endif
+endfunc
+
+func Test_cbuffer_range()
+  call XbufferTests_range('c')
+  "call XbufferTests_range('l')
```
Comment (yegappan) [inline src/testdir/test_quickfix.vim]: Is the location list test failing?

File: src/testdir/test_quickfix.vim [hunk 3]
```diff
@@ -6496,4 +6496,70 @@ func Test_efm_format_b()
...
+  call assert_true(len(l) == 1 &&
+	\ l[0].lnum == 900 && l[0].col == 55 && l[0].text ==# 'Line 900')
+
+  enew!
+  let lines =<< trim END
+    Xtestfile11:700:20:Line 700
+    Xtestfile12:750:25:Line 750
+  END
+  silent! call setline(1, lines)
+  if (a:cchar == 'c')
```
Comment (yegappan) [inline src/testdir/test_quickfix.vim:6543]: Using `Xaddbuffer` here doesn't exercise the range functionality?

**Review by yegappan — CHANGES_REQUESTED** [review]

File: src/testdir/test_quickfix.vim (hunk 1 above)
Comment (chrisbra) [reply src/testdir/test_quickfix.vim:6528]: I initially used `Xbuffer` with `command! -nargs=* -bang -range=% Xbuffer <mods><line1>,<line2>cbuffer<bang> <args>` and then using `Xbuffer` around line '6510' , but unfortunately, then it didn't fail properly, so I resorted back to using the raw `:c/l` commands for testing. Also here on this line.

File: src/testdir/test_quickfix.vim (hunk 2 above)
Comment (chrisbra) [reply src/testdir/test_quickfix.vim]: yes it is, still commented out from debugging :/ I'll push a fix in a second

File: src/testdir/test_quickfix.vim (hunk 3 above)
Comment (chrisbra) [reply src/testdir/test_quickfix.vim:6543]: doesn't fail unfortunately :/

File: src/testdir/test_quickfix.vim [hunk 4]
```diff
@@ -6496,4 +6496,74 @@ func Test_efm_format_b()
...
+  let lines =<< trim END
+    Xtestfile7:700:10:Line 700
+    Xtestfile8:800:15:Line 800
+  END
+  silent! call setline(1, lines)
+  norm! Vy
+  " Note: We cannot use :Xbuffer here,
+  " it doesn't properly fail, so we need to
+  " test using the raw c/l commands.
+  " (also furthre down)
```
Comment (diegoviola) [inline src/testdir/test_quickfix.vim]: @chrisbra typo.

Comment (diegoviola) [reply src/testdir/test_quickfix.vim]: furthre -> further.

Comment (chrisbra) [reply src/testdir/test_quickfix.vim]: thanks, fixed

**Pull request closed without merging**